src/Main.elm

```diff
@@ -1,10 +1,177 @@
+
+
+view : Model -> Html Msg
+view zipper =
+    let
+        (Info { audiences, name, id }) =
+            Zipper.label zipper
+
+        folders =
+            kids zipper
+
+        {- NOTE: I would not really write it like this in real project. Just thought I would have some fun and perhaps humor someone with a pointfree eye ;) -}
+        ifNotRoot =
+            (flip Maybe.andThen <| id) << always << Just
+
+        {- NOTE: I decided against it. One button is enough. -}
+        -- rootButton = viewGoButton "Back to TOP" <| Zipper.root zipper
+        -- toTop = ifNotRoot rootButton
+        goUp =
+            ifNotRoot <| viewGoUp zipper
+
+        folderTitle =
+            Html.span
+                [ style "padding" "15px"
+                , style "margin" "15px"
+                , style "background-color" "pink"
+                ]
+                [ Html.text <| "Now viewing: " ++ name ]
+
+        levelName =
+            ifNotRoot folderTitle
+
+        directories =
+            viewFolders folders
+
+        items =
+            if List.isEmpty audiences then
+                [{- EMPTY FOLDER -}]
+
+            else
+                List.map viewAudience audiences
+
+        navigation =
+            Maybe.Extra.values
+                [ -- toTop
+                  goUp
+                , levelName
+                ]
+    in
+    Html.div
+        [ style "margin" "auto"
+        , style "margin-top" "0"
+        , style "margin-bottom" "0"
+        , style "width" "50%"
+        , style "border" "3px solid green"
+        ]
+    <|
+        navigation
+            ++ [ Html.ul [ style "list-style-type" "none", style "overflow-y" "scroll", style "height" "90vh" ] <|
+                    directories
+                        ++ items
+               ]
+
+
+viewFolders : List (Zipper FolderInfo) -> List (Html Msg)
+viewFolders folders =
+    List.map viewFolder folders
+
+
+viewGoButton : String -> Zipper FolderInfo -> Html Msg
+viewGoButton message goWhere =
+    Html.div
+        [ Events.onClick <| Navigate goWhere
+        , style "background-color" "teal"
+        , style "padding" "20px"
+        ]
+        [ Html.text message ]
+
+
+viewGoUp : Zipper FolderInfo -> Html Msg
+viewGoUp zipper =
+    let
+        maybeParent =
+            Zipper.parent zipper
+
+        {- NOTE: Honestly, I am not happy with the code here. I think some more thought out strategy might help.
+           Like - since we know this function is only called when it IS POSSIBLE to go up, we might leverage that and shift the responsibility
+           of proving that the current level HAS a parent to the caller.
+           But I have spent enought time with this code already so fine tuning it seems like a pointless task - epsecially since I can just write my thoughts here.
+        -}
+        goMessage =
+            Maybe.andThen
+                (Zipper.label >> (\(Info { name }) -> String.Extra.nonEmpty name))
```

Comment:
Not sure, if this `nonEmpty` constrain for name is necessary. You should probably trust API in this case. 
Also you are extracting `Zipper.parent zipper` for two cases differently. 
Imagine case where you have parent but has empty name. What will happen?

Button will lead to `parent` but message will be `"go to root"` which will be misleading.

I would do simple case here like 

```Elm
case Zipper.parent zipper of 
    Just parent ->
        viewGoButton ("go to " ++ doExtractingLabelMagicHere) parent
    Nothing ->
        viewGoButton "go to root" (Zipper.root zipper)
        
```        
     
 This has also advantage with not executing `(Zipper.root zipper)` in every case. In you case this will be always executed even if you don't need it.
 
 Also 
 ```Elm
 |> Maybe.map ((++) "go to ")
 |> Maybe.Extra.unwrap "go to root" identity
 ```
 
 is basically 
 
  ```Elm
 |> Maybe.Extra.unwrap "go to root" ((++) "go to ")
 ```

Reply:
You are right. This was my major pain point. I knew it just ain't it, but was too tired/stuck. I should have given it some time and see it from a different perspective, but I wanted to submit quickly.
It is pretty obvious that I have needlesly overcomplicated the solution and spiraled into the pit. Should have known better.

Regarding the unnecessary `Zipper.root zipper` - honestly I just didn't even question it - and now it kinda seems funny, because it is obvious that Haskell's laziness made me lazy too.

Reply:
The last point is almost embarrassing honestly.
I should never leave it like this, with identity - which is a dead giveaway anyway.
I probably should have given it an extra day and review my code with a fresh set of eyes.
Well, it is a teachable moment.

Reply:
No worries, I just had to find some issues so you do not feel to confident 😉